app/lib/features/news/widgets/news_item.dart

```diff
@@ -169,44 +172,101 @@ class _NewsItemState extends ConsumerState<NewsItem> {
   Widget newsActionButtons({required NewsSlide newsSlide}) {
     final newsReferencesList = newsSlide.references().toList();
     if (newsReferencesList.isEmpty) return const SizedBox();
-
     final referenceDetails = newsReferencesList.first.refDetails();
-    final uriId = referenceDetails.uri() ?? '';
-    final title = referenceDetails.title() ?? '';
-
-    if (title == NewsReferencesType.shareEvent.name) {
-      final calEventLoader = ref.watch(calendarEventProvider(uriId));
-      return calEventLoader.when(
-        data: (calEvent) => EventItem(event: calEvent),
-        loading: () => const EventItemSkeleton(),
-        error: (e, s) {
-          _log.severe('Failed to load cal event', e, s);
-          return Card(
-            child: ListTile(
-              leading: const Icon(Icons.calendar_month),
-              title: Text(L10n.of(context).eventNoLongerAvailable),
-              subtitle: Text(
-                L10n.of(context).eventDeletedOrFailedToLoad,
-                style: Theme.of(context).textTheme.labelLarge,
-              ),
-              onTap: () async {
-                await ActerErrorDialog.show(
-                  context: context,
-                  error: e,
-                  stack: s,
-                );
-              },
-            ),
-          );
-        },
+    return renderActionButton(referenceDetails);
+  }
+
+  Widget renderActionButton(RefDetails referenceDetails) {
+    final evtType = NewsReferencesType.fromStr(referenceDetails.typeStr());
+
+    return switch (evtType) {
+      NewsReferencesType.calendarEvent => renderCalendarEventAction(
+          targetEventId: referenceDetails.targetIdStr() ?? '',),
+      NewsReferencesType.link => renderLinkActionButtion(referenceDetails),
```

Comment:
Not quite understand the use-case of `Link`, is that was for backend support?

Reply:
Yes, that was what we did before - but it was wrong

Reply:
In general Link is for any external URL to anything else.